src/framework/components/collision/component.js

```diff
@@ -315,6 +315,9 @@ class CollisionComponent extends Component {
             if (asset && (!asset.resource || !this.data.shape)) {
                 this.system.recreatePhysicalShapes(this);
                 return;
+            } else if (! this.data.initialized && ! this.data.shape) {
```

Comment:
the code both ifs execute is the same. ideally we refactor the condition and have the code only one time.

Reply:
Right and it is in the wrong place. Let me refactor.